great_expectations/checkpoint/checkpoint.py

```diff
@@ -356,23 +359,22 @@ def run(
         run_id: Optional[Union[str, RunIdentifier]] = None,
         run_name: Optional[str] = None,
         run_time: Optional[Union[str, datetime.datetime]] = None,
-        result_format: Optional[str] = None,
+        result_format: Optional[Union[str, dict]] = None,
         expectation_suite_ge_cloud_id: Optional[str] = None,
-        **kwargs,
     ) -> CheckpointResult:
         assert not (run_id and run_name) and not (
             run_id and run_time
         ), "Please provide either a run_id or run_name and/or run_time."
 
         run_time = run_time or datetime.datetime.now()
-        runtime_configuration: dict = runtime_configuration or {}
-        result_format: Optional[dict] = result_format or runtime_configuration.get(
-            "result_format"
-        )
+        runtime_configuration = runtime_configuration or {}
+        result_format = result_format or runtime_configuration.get("result_format")
```

Comment:
Why no type hint here, but yes on my comment below? What is our convention regarding type hints in function bodies?

Reply:
@NathanFarmer Same reason -- added where there was none before.

Reply:
Sorry if I wasn't clear. Why did you _remove_ the type hint here?

Reply:
@NathanFarmer I removed it, because I saw it was already provided in the arguments.

Reply:
👍🏻